From e11e9dc8298dc0632050cacce96e9652d017f755 Mon Sep 17 00:00:00 2001
From: Varun Reddy Yeturu <varunreddy.yeturu@codeaurora.org>
Date: Tue, 3 Oct 2017 17:11:19 -0700
Subject: qcacld-3.0: Avoid overflow of WMI descriptor pool array

Ensure that the array index of WMI descriptor pool is
within the maximum pool size

Change-Id: I32bf6a9689d3119369cfe2284f51f6592aa00047
CRs-Fixed: 2119803
---
 core/wma/src/wma_mgmt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/wma/src/wma_mgmt.c b/core/wma/src/wma_mgmt.c
index 388bb14..5834cf9 100644
--- a/core/wma/src/wma_mgmt.c
+++ b/core/wma/src/wma_mgmt.c
@@ -2735,6 +2735,10 @@ static int wma_process_mgmt_tx_completion(tp_wma_handle wma_handle,
 		WMA_LOGE("%s: NULL pdev pointer", __func__);
 		return -EINVAL;
 	}
+	if (desc_id >= WMI_DESC_POOL_MAX) {
+		WMA_LOGE("%s: Invalid desc id %d", __func__, desc_id);
+		return -EINVAL;
+	}
 
 	WMA_LOGD("%s: status: %d wmi_desc_id: %d", __func__, status, desc_id);
 
-- 
cgit v1.1

